pkg/extensions/tbtc/derive_address.go

```diff
@@ -0,0 +1,105 @@
+package tbtc
```

Comment:
I think `pkg/extenstions/tbtc/recovery` will contain another higher-level helper function glues a lot of the existing helper functions together, as well as the code that manages the the latest derivation index.

I had left derive_address.go in the `tbtc` directory because it seems more specific to general btc operations than to assisting in recovery directly. I'm very indifferent, so if you have even a mild preference I'll change it right away.

Reply:
As we plan to use it only for recovery and not as a util function across other packages, I'd lean towards placing it inside the `recovery` package.